File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for(auto i : nums) {
            m[i]++;
        }
        priority_queue<pair<int, int>> pq;
        for(auto i : m) {
            pq.push({i.second,i.first});
        }
        vector<int> res;
        for(int i = 0; i < k; i++) {
            pair<int,int> val = pq.top();
            res.push_back(val.second);
            pq.pop();
        }
        return res;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i = 0 ;i < n; i++) {
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) ->
            map.get(b) - map.get(a)
        );

        pq.addAll(map.keySet());
        int[] res = new int[k];
        // int i = -1;
        for(int i = 0 ;i < k; i++) {
            res[i] = pq.poll();
        }
        return res;
    }
}